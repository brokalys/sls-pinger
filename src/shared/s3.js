const { S3Client, PutObjectCommand } = require('@aws-sdk/client-s3');

const client = new S3Client();

module.exports = {
  putObject: function (message) {
    return client.send(new PutObjectCommand(message));
  },
};
