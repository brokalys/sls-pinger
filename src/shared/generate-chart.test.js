import moment from 'moment';
import s3 from './s3';
import generateChart from './generate-chart';

jest.mock('./s3');

describe('generate-chart', () => {
  afterEach(jest.clearAllMocks);

  test('constructs a SVG chart', async () => {
    await generateChart(
      'test.svg',
      [{ x: '2021-01-01', y: 120 }],
      moment('2021-01-12').valueOf(),
    );

    expect(s3.putObject).toBeCalled();
    expect(s3.putObject.mock.calls[0][0].Body).toMatchSnapshot();
  });

  test('uploads the SVG chart to S3', async () => {
    s3.getSignedUrlPromise.mockReturnValue('https://url-to-file/test.svg');

    const url = await generateChart(
      'test.svg',
      [{ x: '2021-01-01', y: 120 }],
      moment('2021-01-12').valueOf(),
    );

    expect(s3.putObject).toBeCalled();
    expect(s3.getSignedUrlPromise).toBeCalled();
    expect(url).toMatch('https://url-to-file/test.svg');
  });
});
