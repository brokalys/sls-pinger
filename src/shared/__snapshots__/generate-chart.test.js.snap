// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate-chart constructs a SVG chart 1`] = `"<svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" version=\\"1.1\\" class=\\"marks\\" width=\\"638\\" height=\\"332\\" viewBox=\\"0 0 638 332\\"><g fill=\\"none\\" stroke-miterlimit=\\"10\\" transform=\\"translate(30,10)\\"><g class=\\"mark-group role-frame root\\" role=\\"graphics-object\\" aria-roledescription=\\"group mark container\\"><g transform=\\"translate(0,0)\\"><path class=\\"background\\" aria-hidden=\\"true\\" d=\\"M0,0h600v300h-600Z\\"/><g><g class=\\"mark-group role-axis\\" role=\\"graphics-symbol\\" aria-roledescription=\\"axis\\" aria-label=\\"Y-axis titled 'Cena m2' for a linear scale with values from 0 to 120\\"><g transform=\\"translate(0.5,0.5)\\"><path class=\\"background\\" aria-hidden=\\"true\\" d=\\"M0,0h0v0h0Z\\" pointer-events=\\"none\\"/><g><g class=\\"mark-rule role-axis-tick\\" pointer-events=\\"none\\"><line transform=\\"translate(0,300)\\" x2=\\"-5\\" y2=\\"0\\" stroke=\\"#888\\" stroke-width=\\"1\\" opacity=\\"1\\"/><line transform=\\"translate(0,275)\\" x2=\\"-5\\" y2=\\"0\\" stroke=\\"#888\\" stroke-width=\\"1\\" opacity=\\"1\\"/><line transform=\\"translate(0,250)\\" x2=\\"-5\\" y2=\\"0\\" stroke=\\"#888\\" stroke-width=\\"1\\" opacity=\\"1\\"/><line transform=\\"translate(0,225)\\" x2=\\"-5\\" y2=\\"0\\" stroke=\\"#888\\" stroke-width=\\"1\\" opacity=\\"1\\"/><line transform=\\"translate(0,200)\\" x2=\\"-5\\" y2=\\"0\\" stroke=\\"#888\\" stroke-width=\\"1\\" opacity=\\"1\\"/><line transform=\\"translate(0,175)\\" x2=\\"-5\\" y2=\\"0\\" stroke=\\"#888\\" stroke-width=\\"1\\" opacity=\\"1\\"/><line transform=\\"translate(0,150)\\" x2=\\"-5\\" y2=\\"0\\" stroke=\\"#888\\" stroke-width=\\"1\\" opacity=\\"1\\"/><line transform=\\"translate(0,125)\\" x2=\\"-5\\" y2=\\"0\\" stroke=\\"#888\\" stroke-width=\\"1\\" opacity=\\"1\\"/><line transform=\\"translate(0,100)\\" x2=\\"-5\\" y2=\\"0\\" stroke=\\"#888\\" stroke-width=\\"1\\" opacity=\\"1\\"/><line transform=\\"translate(0,75)\\" x2=\\"-5\\" y2=\\"0\\" stroke=\\"#888\\" stroke-width=\\"1\\" opacity=\\"1\\"/><line transform=\\"translate(0,50)\\" x2=\\"-5\\" y2=\\"0\\" stroke=\\"#888\\" stroke-width=\\"1\\" opacity=\\"1\\"/><line transform=\\"translate(0,25)\\" x2=\\"-5\\" y2=\\"0\\" stroke=\\"#888\\" stroke-width=\\"1\\" opacity=\\"1\\"/><line transform=\\"translate(0,0)\\" x2=\\"-5\\" y2=\\"0\\" stroke=\\"#888\\" stroke-width=\\"1\\" opacity=\\"1\\"/></g><g class=\\"mark-text role-axis-label\\" pointer-events=\\"none\\"><text text-anchor=\\"end\\" transform=\\"translate(-7,303)\\" font-family=\\"sans-serif\\" font-size=\\"10px\\" fill=\\"#000\\" opacity=\\"1\\">0</text><text text-anchor=\\"end\\" transform=\\"translate(-7,278)\\" font-family=\\"sans-serif\\" font-size=\\"10px\\" fill=\\"#000\\" opacity=\\"1\\">10</text><text text-anchor=\\"end\\" transform=\\"translate(-7,253)\\" font-family=\\"sans-serif\\" font-size=\\"10px\\" fill=\\"#000\\" opacity=\\"1\\">20</text><text text-anchor=\\"end\\" transform=\\"translate(-7,228)\\" font-family=\\"sans-serif\\" font-size=\\"10px\\" fill=\\"#000\\" opacity=\\"1\\">30</text><text text-anchor=\\"end\\" transform=\\"translate(-7,203.00000000000003)\\" font-family=\\"sans-serif\\" font-size=\\"10px\\" fill=\\"#000\\" opacity=\\"1\\">40</text><text text-anchor=\\"end\\" transform=\\"translate(-7,177.99999999999997)\\" font-family=\\"sans-serif\\" font-size=\\"10px\\" fill=\\"#000\\" opacity=\\"1\\">50</text><text text-anchor=\\"end\\" transform=\\"translate(-7,153)\\" font-family=\\"sans-serif\\" font-size=\\"10px\\" fill=\\"#000\\" opacity=\\"1\\">60</text><text text-anchor=\\"end\\" transform=\\"translate(-7,127.99999999999999)\\" font-family=\\"sans-serif\\" font-size=\\"10px\\" fill=\\"#000\\" opacity=\\"1\\">70</text><text text-anchor=\\"end\\" transform=\\"translate(-7,103.00000000000001)\\" font-family=\\"sans-serif\\" font-size=\\"10px\\" fill=\\"#000\\" opacity=\\"1\\">80</text><text text-anchor=\\"end\\" transform=\\"translate(-7,78)\\" font-family=\\"sans-serif\\" font-size=\\"10px\\" fill=\\"#000\\" opacity=\\"1\\">90</text><text text-anchor=\\"end\\" transform=\\"translate(-7,52.999999999999986)\\" font-family=\\"sans-serif\\" font-size=\\"10px\\" fill=\\"#000\\" opacity=\\"1\\">100</text><text text-anchor=\\"end\\" transform=\\"translate(-7,28.00000000000001)\\" font-family=\\"sans-serif\\" font-size=\\"10px\\" fill=\\"#000\\" opacity=\\"1\\">110</text><text text-anchor=\\"end\\" transform=\\"translate(-7,3)\\" font-family=\\"sans-serif\\" font-size=\\"10px\\" fill=\\"#000\\" opacity=\\"1\\">120</text></g><g class=\\"mark-rule role-axis-domain\\" pointer-events=\\"none\\"><line transform=\\"translate(0,300)\\" x2=\\"0\\" y2=\\"-300\\" stroke=\\"#888\\" stroke-width=\\"1\\" opacity=\\"1\\"/></g><g class=\\"mark-text role-axis-title\\" pointer-events=\\"none\\"><text text-anchor=\\"middle\\" transform=\\"translate(-14,150) rotate(-90) translate(0,-2)\\" font-family=\\"sans-serif\\" font-size=\\"11px\\" font-weight=\\"bold\\" fill=\\"#000\\" opacity=\\"1\\">Cena m2</text></g></g><path class=\\"foreground\\" aria-hidden=\\"true\\" d=\\"\\" pointer-events=\\"none\\" display=\\"none\\"/></g></g><g class=\\"mark-group role-axis\\" role=\\"graphics-symbol\\" aria-roledescription=\\"axis\\" aria-label=\\"X-axis for a time scale with values from 01-01 to 01-12\\"><g transform=\\"translate(0.5,300.5)\\"><path class=\\"background\\" aria-hidden=\\"true\\" d=\\"M0,0h0v0h0Z\\" pointer-events=\\"none\\"/><g><g class=\\"mark-rule role-axis-tick\\" pointer-events=\\"none\\"><line transform=\\"translate(0,0)\\" x2=\\"0\\" y2=\\"5\\" stroke=\\"#888\\" stroke-width=\\"1\\" opacity=\\"1\\"/><line transform=\\"translate(55,0)\\" x2=\\"0\\" y2=\\"5\\" stroke=\\"#888\\" stroke-width=\\"1\\" opacity=\\"1\\"/><line transform=\\"translate(109,0)\\" x2=\\"0\\" y2=\\"5\\" stroke=\\"#888\\" stroke-width=\\"1\\" opacity=\\"1\\"/><line transform=\\"translate(164,0)\\" x2=\\"0\\" y2=\\"5\\" stroke=\\"#888\\" stroke-width=\\"1\\" opacity=\\"1\\"/><line transform=\\"translate(218,0)\\" x2=\\"0\\" y2=\\"5\\" stroke=\\"#888\\" stroke-width=\\"1\\" opacity=\\"1\\"/><line transform=\\"translate(273,0)\\" x2=\\"0\\" y2=\\"5\\" stroke=\\"#888\\" stroke-width=\\"1\\" opacity=\\"1\\"/><line transform=\\"translate(327,0)\\" x2=\\"0\\" y2=\\"5\\" stroke=\\"#888\\" stroke-width=\\"1\\" opacity=\\"1\\"/><line transform=\\"translate(382,0)\\" x2=\\"0\\" y2=\\"5\\" stroke=\\"#888\\" stroke-width=\\"1\\" opacity=\\"1\\"/><line transform=\\"translate(436,0)\\" x2=\\"0\\" y2=\\"5\\" stroke=\\"#888\\" stroke-width=\\"1\\" opacity=\\"1\\"/><line transform=\\"translate(491,0)\\" x2=\\"0\\" y2=\\"5\\" stroke=\\"#888\\" stroke-width=\\"1\\" opacity=\\"1\\"/><line transform=\\"translate(545,0)\\" x2=\\"0\\" y2=\\"5\\" stroke=\\"#888\\" stroke-width=\\"1\\" opacity=\\"1\\"/><line transform=\\"translate(600,0)\\" x2=\\"0\\" y2=\\"5\\" stroke=\\"#888\\" stroke-width=\\"1\\" opacity=\\"1\\"/></g><g class=\\"mark-text role-axis-label\\" pointer-events=\\"none\\"><text text-anchor=\\"middle\\" transform=\\"translate(0,15)\\" font-family=\\"sans-serif\\" font-size=\\"10px\\" fill=\\"#000\\" opacity=\\"1\\">01-01</text><text text-anchor=\\"middle\\" transform=\\"translate(54.54545454545455,15)\\" font-family=\\"sans-serif\\" font-size=\\"10px\\" fill=\\"#000\\" opacity=\\"1\\">01-02</text><text text-anchor=\\"middle\\" transform=\\"translate(109.0909090909091,15)\\" font-family=\\"sans-serif\\" font-size=\\"10px\\" fill=\\"#000\\" opacity=\\"1\\">01-03</text><text text-anchor=\\"middle\\" transform=\\"translate(163.63636363636363,15)\\" font-family=\\"sans-serif\\" font-size=\\"10px\\" fill=\\"#000\\" opacity=\\"1\\">01-04</text><text text-anchor=\\"middle\\" transform=\\"translate(218.1818181818182,15)\\" font-family=\\"sans-serif\\" font-size=\\"10px\\" fill=\\"#000\\" opacity=\\"1\\">01-05</text><text text-anchor=\\"middle\\" transform=\\"translate(272.7272727272727,15)\\" font-family=\\"sans-serif\\" font-size=\\"10px\\" fill=\\"#000\\" opacity=\\"1\\">01-06</text><text text-anchor=\\"middle\\" transform=\\"translate(327.27272727272725,15)\\" font-family=\\"sans-serif\\" font-size=\\"10px\\" fill=\\"#000\\" opacity=\\"1\\">01-07</text><text text-anchor=\\"middle\\" transform=\\"translate(381.8181818181818,15)\\" font-family=\\"sans-serif\\" font-size=\\"10px\\" fill=\\"#000\\" opacity=\\"1\\">01-08</text><text text-anchor=\\"middle\\" transform=\\"translate(436.3636363636364,15)\\" font-family=\\"sans-serif\\" font-size=\\"10px\\" fill=\\"#000\\" opacity=\\"1\\">01-09</text><text text-anchor=\\"middle\\" transform=\\"translate(490.90909090909093,15)\\" font-family=\\"sans-serif\\" font-size=\\"10px\\" fill=\\"#000\\" opacity=\\"1\\">01-10</text><text text-anchor=\\"middle\\" transform=\\"translate(545.4545454545454,15)\\" font-family=\\"sans-serif\\" font-size=\\"10px\\" fill=\\"#000\\" opacity=\\"1\\">01-11</text><text text-anchor=\\"middle\\" transform=\\"translate(600,15)\\" font-family=\\"sans-serif\\" font-size=\\"10px\\" fill=\\"#000\\" opacity=\\"1\\">01-12</text></g><g class=\\"mark-rule role-axis-domain\\" pointer-events=\\"none\\"><line transform=\\"translate(0,0)\\" x2=\\"600\\" y2=\\"0\\" stroke=\\"#888\\" stroke-width=\\"1\\" opacity=\\"1\\"/></g></g><path class=\\"foreground\\" aria-hidden=\\"true\\" d=\\"\\" pointer-events=\\"none\\" display=\\"none\\"/></g></g><g class=\\"mark-symbol role-mark\\" role=\\"graphics-symbol\\" aria-roledescription=\\"symbol mark container\\"><path transform=\\"translate(0,0)\\" d=\\"M3.872983346207417,0A3.872983346207417,3.872983346207417,0,1,1,-3.872983346207417,0A3.872983346207417,3.872983346207417,0,1,1,3.872983346207417,0\\" fill=\\"#4c78a8\\" fill-opacity=\\"0.3\\"/></g><g class=\\"mark-group role-scope\\" role=\\"graphics-object\\" aria-roledescription=\\"group mark container\\"><g transform=\\"translate(0,0)\\"><path class=\\"background\\" aria-hidden=\\"true\\" d=\\"M0,0h0v0h0Z\\"/><g><g class=\\"mark-line role-mark\\" role=\\"graphics-symbol\\" aria-roledescription=\\"line mark container\\"><path d=\\"M0,0Z\\" stroke=\\"firebrick\\" stroke-width=\\"2\\"/></g></g><path class=\\"foreground\\" aria-hidden=\\"true\\" d=\\"\\" display=\\"none\\"/></g></g></g><path class=\\"foreground\\" aria-hidden=\\"true\\" d=\\"\\" display=\\"none\\"/></g></g></g></svg>"`;
